desc/geometry/curve.py

```diff
@@ -895,6 +902,375 @@ def from_values(cls, coords, N=10, basis="xyz", name=""):
+        ----------
+        names : str or array-like of str
+            Name(s) of the quantity(s) to compute.
+        grid : Grid or int, optional
+            Grid of coordinates to evaluate at. Defaults to a Linear grid.
+            If an integer, uses that many equally spaced points.
+        params : dict of ndarray
+            Parameters from the equilibrium. Defaults to attributes of self.
+        transforms : dict of Transform
+            Transforms for R, Z, lambda, etc. Default is to build from grid
+        data : dict of ndarray
+            Data computed so far, generally output from other compute functions.
+            Any vector v = v¹ R̂ + v² ϕ̂ + v³ Ẑ should be given in components
+            v = [v¹, v², v³] where R̂, ϕ̂, Ẑ are the normalized basis vectors
+            of the cylindrical coordinates R, ϕ, Z.
+        override_grid : bool
+            If True, override the user supplied grid if necessary and use a full
+            resolution grid to compute quantities and then downsample to user requested
+            grid. If False, uses only the user specified grid, which may lead to
+            inaccurate values for surface or volume averages.
+
+        Returns
+        -------
+        data : dict of ndarray
+            Computed quantity and intermediate variables.
+
+        """
+        return super().compute(
+            names=names,
+            grid=grid,
+            params=params,
+            transforms=transforms,
+            data=data,
+            override_grid=override_grid,
+            basis_in=self.basis,
+            **kwargs,
+        )
+
+    @classmethod
+    def from_values(cls, coords, N=10, s=None, basis="xyz", name=""):
+        """Fit coordinates to FourierXYCurve representation.
+
+        Parameters
+        ----------
+        coords: ndarray, shape (num_coords,3)
+            Coordinates to fit a FourierXYCurve object with each column
+            corresponding to xyz or rpz depending on the basis argument.
+        N : int
+            Fourier resolution of the new X & Y representation.
+        s : ndarray or "arclength"
+            Arbitrary curve parameter to use for the fitting.
+            Should be monotonic, 1D array of same length as
+            coords. if None, defaults linearly spaced in [0,2pi)
+            Alternative, can pass "arclength" to use normalized distance between points.
+        basis : {"rpz", "xyz"}
+            Basis for input coordinates. Defaults to "xyz".
+        name : str
+            Name for this curve.
+
+        Returns
+        -------
+        curve : FourierXYCurve
+            New representation of the curve parameterized by Fourier series for X and Y.
+
+        """
+        # convert to xyz basis
+        if basis == "rpz":
+            coords = rpz2xyz(coords)
+        coords = np.atleast_2d(coords)
+
+        # center
+        center = np.mean(coords, axis=0)
+        coords = coords - center  # shift to origin
+
+        # normal
```

Comment:
Question for this curve class: 

technically it works totally fine if the curve orientation and the normal direction are opposite (in the sense that you can have a curve which is oriented left-handed w.r.t. the normal vector). Unless we include a line like Ryan has in the `from_values` of `FourierPlanarCurve` this will be allowed. This does not break anything as far as I can tell. Just want to note this as could be slightly counterintuitive

Reply:
Hmm good point. There is a "right-handed" orientation wrt the normal vector, but it doesn't actually affect any of the computations. The only thing that matters is the physical direction of the current flowing in the coil. I think we should keep it general so that the user can choose the orientation they find the most convenient (e.g., they might want a "left-handed" curve if that allows them to make the current positive instead of negative.)